src/pages/lib/graphqlapi/subscribe-data/q/platform/[platform].mdx

```diff
@@ -2,3 +2,455 @@ export const meta = {
+- If you are using dynamic group authorization and you authorize based on a single group per record, then subscriptions are only supported if the user is part of 5 or fewer user groups.
+- Additionally, if you authorize via an array of groups (`groups: [String]`),
+  - subscriptions are only supported if the user is part of 20 or fewer groups
+  - you can only authorize 20 or fewer user groups per record
+
+</Callout>
+
+### Subscription connection status updates
+
+Now that your application is set up and using subscriptions, you may want to know when the subscription is finally established, or reflect to your users when the subscription isn't healthy. You can monitor the connection state for changes through the `Hub` local eventing system.
+
+```ts
+import { CONNECTION_STATE_CHANGE, ConnectionState } from '@aws-amplify/pubsub';
+import { Hub } from 'aws-amplify';
+
+Hub.listen('api', (data: any) => {
+  const { payload } = data;
+  if (payload.event === CONNECTION_STATE_CHANGE) {
+    const connectionState = payload.data.connectionState as ConnectionState;
+    console.log(connectionState);
+  }
+});
+```
+
+#### Connection states
+
+- **`Connected`** - Connected and working with no issues.
+- **`ConnectedPendingDisconnect`** - The connection has no active subscriptions and is disconnecting.
+- **`ConnectedPendingKeepAlive`** - The connection is open, but has missed expected keep-alive messages.
+- **`ConnectedPendingNetwork`** - The connection is open, but the network connection has been disrupted. When the network recovers, the connection will continue serving traffic.
+- **`Connecting`** - Attempting to connect.
+- **`ConnectionDisrupted`** - The connection is disrupted and the network is available.
+- **`ConnectionDisruptedPendingNetwork`** - The connection is disrupted and the network connection is unavailable.
+- **`Disconnected`** - Connection has no active subscriptions and is disconnecting.
+
+<Accordion title='Troubleshooting connection issues and automated reconnection' headingLevel='4' eyebrow='Troubleshooting'>
+
+Connections between your application and backend subscriptions can be interrupted for various reasons, including network outages or the device entering sleep mode. Your subscriptions will automatically reconnect when it becomes possible to do so.
+
+While offline, your application will miss messages and will not automatically catch up when reconnection happens. Depending on your use case, you may want to take action for your app to catch up when it comes back online.
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+// ...
+import { GraphQLQuery, GraphQLSubscription } from '@aws-amplify/api';
+import {
+  ListTodosQuery
+  OnCreateTodoSubscription,
+  OnUpdateTodoSubscription,
+  OnDeleteTodoSubscription
+} from './API';
+
+const fetchRecentData = () => {
+  // Retrieve some/all data from AppSync
+  const allTodos = await API.graphql<GraphQLQuery<ListTodosQuery>>({ 
```

Comment:
@renebrandel Line 235 "TODO for me to change all the graphqlOperations patterns" Kept this as I was not sure if you caught them all in your review. So reminder JIC